src/baseline.rs

```diff
@@ -7,22 +7,232 @@ use crate::dump::RustDocCommand;
+                    // give us the latest semver-compatible version which is not we want.
+                    // Fixes: https://github.com/obi1kenobi/cargo-semver-checks/issues/261
+                    version: Some(format!("={}", crate_.version())),
+                    features: crate_features::get_all_crate_features_from_registry(crate_),
+                    ..DependencyDetail::default()
+                },
+                CrateSource::ManifestPath { path, .. } => DependencyDetail {
+                    path: Some(
+                        // The manifest will be saved in some other directory,
+                        // so for convenience, we're using absolute paths.
+                        path.canonicalize()
+                            .context("failed to canonicalize manifest path")?
+                            .to_str()
+                            .context("manifest path is not valid UTF-8")?
+                            .to_string(),
+                    ),
+                    features: crate_features::get_all_crate_features_from_manifest(path),
+                    ..DependencyDetail::default()
+                },
+            };
+            let mut deps = DepsSet::new();
+            deps.insert(
+                match crate_source {
+                    CrateSource::Registry { crate_ } => crate_.name().to_string(),
+                    CrateSource::ManifestPath { name, path: _ } => name.to_string(),
+                },
+                Dependency::Detailed(project_with_features),
+            );
+            deps
+        },
+        ..Default::default()
+    })
+}
+
+fn save_placeholder_rustdoc_manifest(
+    placeholder_build_dir: &Path,
+    placeholder_manifest: cargo_toml::Manifest<()>,
+) -> anyhow::Result<PathBuf> {
+    std::fs::create_dir_all(placeholder_build_dir).context("failed to create build dir")?;
+    let placeholder_manifest_path = placeholder_build_dir.join("Cargo.toml");
+
+    // Possibly fixes https://github.com/libp2p/rust-libp2p/pull/2647#issuecomment-1280221217
+    let _: std::io::Result<()> = std::fs::remove_file(placeholder_build_dir.join("Cargo.lock"));
+
+    std::fs::write(
+        &placeholder_manifest_path,
+        toml::to_string(&placeholder_manifest)?,
+    )
+    .context("failed to write placeholder manifest")?;
+    std::fs::write(placeholder_build_dir.join("lib.rs"), "")
+        .context("failed to create empty lib.rs")?;
+    Ok(placeholder_manifest_path)
+}
+
+fn generate_rustdoc(
+    config: &mut GlobalConfig,
+    rustdoc: &RustDocCommand,
+    target_root: PathBuf,
+    crate_source: CrateSource,
+) -> anyhow::Result<PathBuf> {
+    let name = crate_source.get_name();
+    let version = crate_source.get_version();
+
+    let (build_dir, cache_dir, cached_rustdoc) = match crate_source {
+        CrateSource::Registry { .. } => {
+            let crate_identifier = format!("registry-{}-{}", slugify(&name), slugify(&version));
```

Comment:
```suggestion
            let crate_identifier = format!("registry-{}-{}", slugify(name), slugify(version));
```
This is the place where we use those values without caring whether they are owned or not. We just saved two copies and allocations!